inc/drivers/MicroBitQuadratureDecoder.h

```diff
@@ -0,0 +1,174 @@
+/*
+The MIT License (MIT)
+
+Copyright (c) 2016-2017 Simon Hosie
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+*/
+
+#ifndef MICROBIT_QUADRATUREDECODER_H
+#define MICROBIT_QUADRATUREDECODER_H
+
+#include "mbed.h"
+#include "MicroBitConfig.h"
+#include "MicroBitComponent.h"
+#include "MicroBitPin.h"
+
+                                                        // Configuration flags...
+#define QDEC_USING_SYSTEM_TICK              0x01        // Use systemTick() to keep position up to date.
+#define QDEC_USING_DEBOUNCE                 0x02        // Use input debounce feature.
+#define QDEC_LED_ACTIVE_LOW                 0x04        // Drive LED pin low to activate.
+
+/**
+  * Class definition for MicroBit Quadrature decoder.
+  *
+  */
+class MicroBitQuadratureDecoder : public MicroBitComponent
+{
+    protected:
+    static MicroBitPin pinNC;           // A no-connect pin -- the default for unused LED arguments.
```

Comment:
I was thinking then that multiple driver instances would then imply multiple `pinNC` instances; and that the linker would throw it away if it wasn't used.  However, leaving it to the linker does mean that the object lives forever even if this driver is only used sometimes (if it's never used then the object should vanish).

How big is it?  Is it possible to put it in ROM?  In ROM it's only as bad as all the other code that is present because it'll sometimes be used.